ldelf/ta_elf_rel.c

```diff
@@ -312,28 +329,31 @@ static void e64_relocate(struct ta_elf *elf, unsigned int rel_sidx)
 	}
 
 	/* Check the address is inside TA memory */
-	assert(shdr[rel_sidx].sh_addr < (elf->max_addr - elf->load_addr));
+	if (ADD_OVERFLOW(shdr[rel_sidx].sh_addr,
+			 shdr[rel_sidx].sh_size, &sh_end))
+		err(TEE_ERROR_SECURITY, "Overflow");
+	if (sh_end >= (elf->max_addr - elf->load_addr))
+		err(TEE_ERROR_GENERIC, "Relocation table out of range");
 	rela = (Elf64_Rela *)(elf->load_addr + shdr[rel_sidx].sh_addr);
 
-	/* Check the address is inside TA memory */
-	if (ADD_OVERFLOW(shdr[rel_sidx].sh_addr, shdr[rel_sidx].sh_size,
-			 &sh_end))
-		err(TEE_ERROR_SECURITY, "Overflow");
-	assert(sh_end < (elf->max_addr - elf->load_addr));
 	rela_end = rela + shdr[rel_sidx].sh_size / sizeof(Elf64_Rela);
 	for (; rela < rela_end; rela++) {
 		Elf64_Addr *where = NULL;
 		size_t sym_idx = 0;
 
 		/* Check the address is inside TA memory */
```

Comment:
very minor: maybe can remove some inline comments as this one since the explicit strings used with `err(...);`.

Reply:
There's some overlap with the error message and the comment, but I still think the comments are justified. They could perhaps be made more precise, but that's better handled in another PR.